web/server/routers/post.ts

```diff
@@ -80,6 +79,16 @@ const postFilterSchema = z.object({
   spayNeuterStatus: z.nativeEnum(Status).optional(),
 });
 
+const goodWithMap: Record<GoodWith, string> = {
+  [GoodWith.Cats]: "getsAlongWithCats" as const,
+  [GoodWith.LargeDogs]: "getsAlongWithLargeDogs" as const,
+  [GoodWith.Men]: "getsAlongWithMen" as const,
+  [GoodWith.OlderChildren]: "getsAlongWithOlderKids" as const,
+  [GoodWith.SmallDogs]: "getsAlongWithSmallDogs" as const,
+  [GoodWith.Women]: "getsAlongWithWomen" as const,
+  [GoodWith.YoungChildren]: "getsAlongWithYoungKids" as const,
+};
```

Comment:
Whoops, I should have specified - have the `as const` assertion applied to the map itself like this:
```js
const map = {
  key1: 1,
  key2: 2
} as const;
```